plugins/linux-v4l2/v4l2-output.c

```diff
@@ -199,7 +199,12 @@ static bool virtualcam_start(void *data)
 	for (int i = 0; i < n; i++) {
 		char device[32];
 
-		snprintf(device, 32, "/dev/%s", list[i]->d_name);
+		// Use the return value of snprintf to prevent truncation warning
+		int written = snprintf(device, 32, "/dev/%s", list[i]->d_name);
+		if (written > 32)
```

Comment:
snprintf will never return a value greater than the limit passed in. Which reveals a minor bug. 

`device` should be a 33 character long array in order to include the null terminal (or we should limit snprintf to 31 characters). `device` should also be initialized to ensure that the unwritten bytes are 0 (`char device[32 ] = {0}` or an equivalent memset).

This check should then be >= 32 (or 31 if you choose that).

Reply:
char array for VAAPI and V4L2 are now initialized with `{0}` and `>=32` is used.